src/cpp/session/SessionConsoleProcess.cpp

```diff
@@ -234,7 +235,21 @@ void ConsoleProcess::commonInit()
                            options_.smartTerminal ? core::system::kSmartTerm :
                                                     core::system::kDumbTerm);
 
+      // environment variables that can be used when configuring custom shells
       core::system::setenv(&(options_.environment.get()), "RSTUDIO_TERM", procInfo_->getHandle());
+
+      core::system::setenv(&(options_.environment.get()), "RSTUDIO_PROJ_NAME",
+                           projects::projectContext().file().stem());
+
+      FilePath projPath = core::system::userHomePath();
```

Comment:
The user home path isn't necessarily the default working path when no project is open (see http://docs.rstudio.com/ide/server-pro/r-sessions.html#working-directories) -- is that important for your use case here or do you just need a safe fallback?